src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -0,0 +1,221 @@
+from proto.tbots_software_msgs_pb2 import Vision
+from proto.validation_pb2 import ValidationGeometry, ValidationProto, ValidationStatus
+from proto.vision_pb2 import BallState, RobotState
+from proto.world_pb2 import SimulatorTick, World, WorldState
+from software.constants import SECONDS_TO_MS
+from pyqtgraph.Qt import QtCore, QtGui
+
+from software.networking.threaded_unix_sender import ThreadedUnixSender
+from software.simulated_tests.robot_enters_region import RobotEntersRegion
+
+from software.simulated_tests import validation
+from software.simulated_tests.full_system import FullSystem
+from software.simulated_tests.er_force_simulator import ErForceSimulator
+from software.thunderscope.thunderscope import Thunderscope
+
+from software.logger.logger import createLogger
+
+logger = createLogger(__name__)
+
+PROCESS_BUFFER_DELAY_S = 0.01
+PAUSE_AFTER_FAIL_DELAY_S = 3
+
+
+class TacticTestRunner(object):
+
+    """Run a tactic"""
+
+    def __init__(
+        self, launch_delay_s=0.1, enable_thunderscope=True, runtime_dir="/tmp/tbots"
+    ):
+        """Initialize the TacticTestRunner
+
+        :param launch_delay_s: How long to wait after launching the processes
+        :param enable_thunderscope: If true, thunderscope opens and the test runs
+                                  in realtime
+        :param runtime_dir: Directory to open sockets, store logs and any output files
+
+        """
+
+        # Setup runtime directory
+        try:
+            os.mkdir(runtime_dir)
+        except:
+            pass
+
+        self.enable_thunderscope = enable_thunderscope
+
+        if self.enable_thunderscope:
+            self.thunderscope = Thunderscope()
+            self.thunderscope.configure_default_layout()
+            self.validation_sender = ThreadedUnixSender(runtime_dir + "/validation")
+
+        self.simulator = ErForceSimulator()
+        self.yellow_full_system = FullSystem()
+        time.sleep(launch_delay_s)
+
+        self.last_exception = None
+
+    def run_test(
+        self,
+        always_validation_sequence_set=[[]],
+        eventually_validation_sequence_set=[[]],
+        test_timeout_s=3,
+        tick_duration_s=0.0166,  # Default to 60hz
+    ):
+        """Run a test
+
+        :param always_validation_sequence_set: Validation functions that should
+                                hold on every tick
+        :param eventually_validation_sequence_set: Validation that should
+                                eventually be true, before the test ends
+        :param test_timeout_s: The timeout for the test, if any eventually_validations
+                                remain after the timeout, the test fails.
+        :param tick_duration_s: The simulation step duration
+
+        """
+
+        def __stopper(delay=PROCESS_BUFFER_DELAY_S):
+            """Stop running the test
+
+            :param delay: How long to wait before closing everything, defaults
```

Comment:
Would this benefit from having a minimum buffer delay?

Reply:
im not sure I follow, would this be instead of PROCESS_BUFFER_DELAY_S?

Reply:
Probably closer to a check such as:
```python
delay = MIN_PROCESS_BUFFER_DELAY_S if delay < MIN_PROCESS_BUFFER_DELAY_S else delay
```
Don't know if it's needed, just a suggestion if having too short of a delay wouldn't give enough time before closing anything for whatever reason.